File: java/T628.java
import java.util.Arrays;

/**
 * @Author: jesse
 * @Date: 2021/1/20 2:16 下午
 * 给定一个整型数组，在数组中找出由三个数组成的最大乘积，并输出这个乘积。
 */
public class T628 {
    public int maximumProduct(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length;
        return Math.max(nums[0] * nums[1] * nums[n - 1], nums[n - 1] * nums[n - 2] * nums[n - 3]);
    }
}

File: rust/src/t628.rs
use std::cmp::max;

pub fn maximum_product(mut nums: Vec<i32>) -> i32 {
    nums.sort();
    let n = nums.len();
    max(nums[0] * nums[1] * nums[n - 1], nums[n - 1] * nums[n - 2] * nums[n - 3])
}